cli/src/commands/git/push.rs

```diff
@@ -402,15 +402,35 @@ fn validate_commits_ready_to_push(
         if commit.has_conflict()? {
             reasons.push("it has conflicts");
         }
-        if !args.allow_private && is_private(commit.id())? {
+        let is_private = is_private(commit.id())?;
+        if !args.allow_private && is_private {
             reasons.push("it is private");
         }
         if !reasons.is_empty() {
-            return Err(user_error(format!(
+            let short = short_commit_hash(commit.id());
+            let mut error = user_error(format!(
                 "Won't push commit {} since {}",
-                short_commit_hash(commit.id()),
+                short,
                 reasons.join(" and ")
-            )));
+            ));
+            if !args.allow_private && is_private {
```

Comment:
nit: isn't it better to show the commit detail for whatever reasons? I think it's useful to know which conflicted commit prevented pushing for example.

Reply:
That seems right to me, and wasn't a particularly noisy diff. I've included this suggestion in my last push as 608bb6ea5e8e2a1b7c4c3dc4379f54fba617c4d0 .

Reply:
The main motive for the static prefix on my side was in case someone has reconfigured their `templates.commit_summary` in such a way that doesn't include enough of the default details, especially the change/commit IDs. For now I've left it alone, fixed the `short` variable and squashed the commit.

Repeating "Won't push commit" would also be fine with me and I'm not at all perturbed if the phrasing gets touched up in follow-up work. I'll think on it for a while in case I can come up something that feels better.

Reply:
Another option is "Rejected commit: " or "Problematic commit: " or something like that. I don't feel strongly either, but I do think "Commit summary: " sounds too generic.

Reply:
I do like "rejected" better here, so I've switched to that. I am mildly hesitant about wording that might give the impression that the *remote* side had a problem with it instead, i.e. tripping GitHub branch protection rules.